docs/tutorials/tokens/erc721.md

```diff
@@ -16,3 +16,339 @@ nav_order: 2
+
+## Sending data with a transfer
+
+All transfers (as well as mint/burn operations) support an optional `message` parameter that contains a broadcast or private
+message to be sent along with the transfer. This message follows the same convention as other FireFly messages, and may be comprised
+of text or blob data, and can provide context, metadata, or other supporting information about the transfer. The message will be
+batched, hashed, and pinned to the primary blockchain as described in [key concepts](../keyconcepts/broadcast.html).
+
+The message ID and hash will also be sent to the token connector as part of the transfer operation, to be written to the token blockchain
+when the transaction is submitted. All recipients of the message will then be able to correlate the message with the token transfer.
+
+`POST` `http://127.0.0.1:5000/api/v1/namespaces/default/tokens/transfers`
+
+### Broadcast message:
+```json
+{
+  "amount": 1,
+  "tokenIndex": "1",
+  "to": "0x07eab7731db665caf02bc92c286f51dea81f923f",
+  "message": {
+    "data": [{
+      "value": "payment for goods"
+    }]
+  }
+}
+```
+
+### Private message:
+```json
+{
+  "amount": 1,
+  "tokenIndex": "1",
+  "to": "0x07eab7731db665caf02bc92c286f51dea81f923f",
+  "message": {
+    "header": {
+      "type": "transfer_private",
+    },
+    "group": {
+      "members": [{
+          "identity": "org_1"
+      }]
+    },
+    "data": [{
+      "value": "payment for goods"
+    }]
+  }
+}
+```
+
+Note that all parties in the network will be able to see the transfer (including the message ID and hash), but only
+the recipients of the message will be able to view the actual message data.
+
+## Burn tokens
+You may burn a token by specifying the token's `tokenIndex`. With a token contract deployed by the default token factory, only the owner of a token may
+burn it, but a different contract may define its own permission model.
+
+`POST` `http://127.0.0.1:5000/api/v1/namespaces/default/tokens/burn`
+
+```json
+{
+  "amount": 1,
+  "tokenIndex": "1"
+}
+```
+
+Other parameters:
+- You must specify a `pool` name if you've created more than one pool
+- You may specify a `key` understood by the connector (i.e. an Ethereum address) if you'd like to use a non-default signing identity
+- You may specify `from` if you'd like to burn tokens from a specific identity (default is the same as `key`)
+
+## Token approvals
+You can also approve other wallets to transfer tokens on your behalf with the `/approvals` API. The important fields in a token approval API request are as follows:
+
+- `approved`: Sets whether another account is allowed to transfer tokens out of this wallet or not. If not specified, will default to `false`.
+- `operator`: The other account that is allowed to transfer tokens out of the wallet specified in the `key` field
+- `config.tokenIndex`: The specific token index within the pool that the operator is allowed to transfer. If `0` or not set, the approval is valid for any token.
```

Comment:
Maybe "all tokens" instead of "any token"? It could possibly be read as applying to only one.